File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        mp = defaultdict(list)
        
        for word in strs:
            box = [0 for _ in range(26)]
            for c in word:
                box[ord(c)-ord('a')]+=1
                
            mp[tuple(box)].append(word)
            
        return [v for v in mp.values()]

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> mp;
        
        for(auto &x: strs){
            string tmp = x;
            sort(begin(tmp),end(tmp));
            mp[tmp].emplace_back(x);
        }
        
        vector<vector<string>> res;
        
        for(auto &x: mp){
            res.emplace_back(x.second);
        }
        
        return res;
    }
};